examples/long_tutorials/handle_drift/plot_handle_drift.py

```diff
@@ -0,0 +1,342 @@
+* **nonrigid_accurate** - A decentralised drift correction (DREDGE), introduced by the Paninski group.
+
+When using motion correction in your analysis, please make sure to
+:ref:`cite the appropriate paper for your chosen method<cite-motion-correction>`.
+
+
+**Now, let's dive into running motion correction with these three
+methods on a simulated dataset.**
+
+"""
+
+# %%
+# -------------------------------------------
+# Setting up and preprocessing the recording
+# -------------------------------------------
+#
+# First, we will import the modules we will need for this tutorial:
+
+import matplotlib.pyplot as plt
+import spikeinterface.full as si
+from spikeinterface.generation.drifting_generator import generate_drifting_recording
+from spikeinterface.preprocessing.motion import motion_options_preset
+from spikeinterface.sortingcomponents.motion_interpolation import correct_motion_on_peaks
+from spikeinterface.widgets import plot_peaks_on_probe
+
+# %% read the file
+# Next, we will generate a synthetic, drifting recording. This recording will
+# have 100 separate units with firing rates randomly distributed between
+# 15 and 25 Hz.
+#
+# We will create a zigzag drift pattern on the recording, starting at
+# 100 seconds and with a peak-to-peak period of 100 seconds (so we will
+# have 9 zigzags through our recording). We also add some non-linearity
+# to the imposed motion.
+
+# %%
+#.. note::
+#    This tutorial can take a long time to run with the default arguments.
+#    If you would like to run this locally, you may want to edit ``num_units``
+#    and ``duration`` to smaller values (e.g. 25 and 100 respectively).
+#
+#    Also note, the below code uses multiprocessing. If you are on Windows, you may
+#    need to place the code within a  ``if __name__ == "__main__":`` block.
+
+
+num_units = 100  # 250 still too many I think!
+duration = 1000
+
+_, raw_recording, _ = generate_drifting_recording(
+    num_units=num_units,
+    duration=duration,
+    generate_sorting_kwargs=dict(firing_rates=(15, 25), refractory_period_ms=4.0),
+    seed=42,
+    generate_displacement_vector_kwargs=dict(motion_list=[
+            dict(
+                drift_mode="zigzag",
+                non_rigid_gradient=None, # 0.1,
+                t_start_drift=int(duration/10),
+                t_end_drift=None,
+                period_s=int(duration/10),
+            ),
+        ],
+    )
+)
+print(raw_recording)
+
+# %%
+# Before performing motion correction, we will **preprocess** the recording
+# with a bandpass filter and a common median reference.
+
+filtered_recording = si.bandpass_filter(raw_recording, freq_min=300.0, freq_max=6000.0)
+preprocessed_recording = si.common_reference(filtered_recording, reference="global", operator="median")
+
+# %%
+#.. warning::
+#    It is better to not whiten the recording before motion estimation, as this
+#    will give a better estimate of the peak locations. Whitening should
+#    be performed after motion correction.
```

Comment:
Maybe instead can add a bit more text here on how motion correction fits into the preprocessing pipeline, and mention it should be done after filtering, CAR but before whitening, rather than a warning box? maybe too extreme for something that is probably an edge case.